tests/unit/core/scan/test_scan_context.py

```diff
@@ -0,0 +1,100 @@
+from pathlib import Path
+from typing import Union
+
+import pytest
+
+from ggshield.core.scan.scan_context import ScanContext
+from ggshield.core.scan.scan_mode import ScanMode
+from tests.repository import Repository
+
+
+class TestScanContextRepositoryURL:
+    def _assert_repo_url_in_headers(
+        self, context: ScanContext, expected_url: Union[Path, str]
+    ):
+        assert context.get_http_headers().get("GGShield-Repository-URL") == str(
+            expected_url
+        )
+
+    def _assert_no_repo_url_in_headers(self, context: ScanContext):
+        assert context.get_http_headers().get("GGShield-Repository-URL") is None
+
+    def test_scan_context_no_repo(
+        self,
+        tmp_path: Path,
+    ):
+        """
+        GIVEN a directory which is not a git repo
+        WHEN passing the local path to the scan context
+        THEN there is no GGShield-Repository-URL in the headers
+        """
+        context = ScanContext(
+            scan_mode=ScanMode.PATH,
+            command_path="ggshield secret scan path",
+            target_path=tmp_path,
+        )
+        self._assert_no_repo_url_in_headers(context)
+
+    def test_scan_context_repository_path(
+        self,
+        tmp_path: Path,
+    ):
+        """
+        GIVEN a repository with a remote (filesystem)
+        WHEN passing the local path to the scan context
+        THEN the remote url is found
+        """
+        remote_repo = Repository.create(tmp_path / "remote", bare=True)
+        local_repo = Repository.clone(remote_repo.path, tmp_path / "local")
+        context = ScanContext(
+            scan_mode=ScanMode.PATH,
+            command_path="ggshield secret scan path",
+            target_path=local_repo.path,
+        )
+        self._assert_repo_url_in_headers(context, remote_repo.path)
+
+    @pytest.mark.parametrize(
+        ("remote_url", "expected_url"),
+        [
+            (
+                "https://user:password@github.com:84/GitGuardian/ggshield.git",
+                "github.com/GitGuardian/ggshield",
+            ),
+            (
+                "https://github.com/GitGuardian/ggshield.git",
+                "github.com/GitGuardian/ggshield",
+            ),
+            (
+                "git@github.com:GitGuardian/ggshield.git",
+                "github.com:GitGuardian/ggshield",
```

Comment:
How difficult would it be to normalize it in order to return github.com**/**GitGuardian/ggshield ?

Reply:
I'm already assuming that repository names cannot contain `:` (this is correct for at least the 4 supported providers), so I'll simply replace occurrences with a `/`